tools/distpartitioning/gloo_wrapper.py

```diff
@@ -59,6 +59,74 @@ def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
     for i in range(world_size):
         dist.scatter(output_tensor_list[i], input_tensor_list if i == rank else [], src=i)
 
+def alltoallv_cpu_data(rank, world_size, input_tensor_list, val_dtype):
+    """
+    Wrapper function to providing the alltoallv functionality by using underlying alltoall
+    messaging primitive
+
+    Parameters:
+    -----------
+    rank : int
+        The rank of current worker
+    world_size : int
+        The size of the entire
+    input_tensor_list : List of tensor
+        The tensors to exchange
+    val_dtype : torch dtype
+        torch dtype which is used to create buffer for receiving messages
```

Comment:
is this input argument necessary? the input tensor list should already has it. you should make sure all input tensors have the same dtype.

Reply:
The reason behind such flexibility is that provide this option to the user of this function to receive an output_tensor_list which is of type val_dtype. This could be different from the dtype of the tensors in the input_tensor_list. 

I agree that this case is not present in the use cases so far. And this might lead to some confusion. 

I am removing the val_dtype for now and assume that the output_tensor_list is always of the type which is same as in the input tensor list. The code is appropriately changed.